src/libopensc/card-iasecc.c

```diff
@@ -57,6 +57,12 @@
 		| SC_ALGORITHM_RSA_HASH_SHA1		\
 		| SC_ALGORITHM_RSA_HASH_SHA256)
 
+#define IASECC_CARD_DEFAULT_CAPS ( 0 			\
+		| SC_CARD_CAP_RNG			\
+		| SC_CARD_CAP_APDU_EXT			\
+		| SC_CARD_CAP_USE_FCI_AC		\
+		| SC_CARD_CAP_ISO7816_PIN_INFO)
```

Comment:
Good point! (I asked myself the same question). The quick answer is that it should work with most cards, since the old code sends exactly the same APDU to query the PIN verification status.

SC_CARD_CAP_ISO7816_PIN_INFO enables the SC_PIN_CMD_GET_INFO support in the ISO 7816 layer, used by the updated driver instead of the old code, with the original code quoted below:
```
iasecc_pin_verify()
{
	...
	rv = iasecc_pin_is_verified(card, &pin_cmd, tries_left);
	...
}

iasecc_pin_is_verified()
{
	...
	pin_cmd = *pin_cmd_data;
	pin_cmd.pin1.data = (unsigned char *)"";
	pin_cmd.pin1.len = 0;

	rv = iasecc_chv_verify(card, &pin_cmd, tries_left);
	...
}
```
Where iasecc_chv_verify is called with a zero-length PIN, making it send a short (APDU case 1) to the card, identical to the one sent with the new code.

Even though this case is (unfortunately) not required from the IAS-ECC v1.01 specifications, it is present in the (unreleased) draft V2 specifications.

But the main point is that the new code should work the same as the old one, just using the ISO 7816 layer for its work.

Reply:
I see, thanks for the clearification.